### PR TITLE
UI review Items part I

### LINKED ISSUE #856: UI & ux Review
Some general items arising from a ui/ux review

**Menu**

Additional request: when clicking a parent, auto expand and navigate to the first menu item. This one is still under discussion.

- [x] highlighting of the parent menu when you are on a submenu page : show the entire parent+children element with a white background (?) and highlight the parent
- [x] when the drawer is collapsed, highlight the parent if you are on a submenu page
- [x] add an 'external link' indication to the docs nav link

**Dashboard**
The charts! can they be dynamic? We discussed.. and wondered if one of them could be, at least to start with. Design suggestions to come?
- [x] Change to single stat items
- [x] When displayed portrait on tablet, the widgets flex badly [insert screenshot here]

**tables**
- [x] Add an alternate row highlight ( perhaps a light grey ) maybe change the hover colour to be primary, or a reduced opacity primary? does that screw with placeholder rows though..?

**Site**
- [ ] ~~The footer text is jammed together ( storename username )~~

**Patients**
- [x] Confusion between patient specific encounters and the encounters nav link. Suggest adding a profile icon to the left of the patient details

**JSON forms**
- [x] Autocomplete - the multiple entry option shows as a white control. Make the background grey, add border radius, change the chip bg colour to white?
- [x] Text - Doesn't show underline on focus like other inputs do. Also, maybe secondary is better than primary, the orange is very close to red

**Other components**
- [x] Autocomplete - highlight the search terms when filtering, perhaps bold the search characters?
- [x] Date picker - is the selected date colour the primary? it doesn't look right

### PR DESCRIPTION
**Pull request by mark-prins**
Fixes #856 for general items. There's a couple to address in the features branch too.

**Menu**

- [ ] highlighting of the parent menu when you are on a submenu page. It now has the parent in **bold** with the whole menu (parent+children) in a container with a box shadow. Shown as a very light grey when active, and transparent otherwise.
- [ ] when the drawer is collapsed, highlight the parent if you are on a submenu page
- [ ] add an 'external link' indication to the docs nav link

**Dashboard**
- [ ] The chart icon has been removed
- [ ] When displayed portrait on tablet, the widgets flex badly - have removed the width constraints which were mainly to accomodate the chart

**tables**
- [ ] Added an alternate row highlight in pale grey. changed the hover colour to a knocked back version of the secondary colour ( the primary looked out of place, particularly with the placeholder blue lines ). Checked with the disabled grey text and the placeholder blue text. 

#### Tables
Screenshot 2022-12-20 at 1 08 56 PM
Screenshot 2022-12-20 at 1 09 03 PM
Screenshot 2022-12-20 at 1 09 36 PM

### External Link
Screenshot 2022-12-20 at 10 08 39 PM
Screenshot 2022-12-20 at 10 09 14 PM

#### Collapsed menu highlights the parent item
Screenshot 2022-12-21 at 10 46 58 AM

#### Selected menu
Screenshot 2022-12-21 at 11 57 13 AM

#### Replenishment selected, hover over Catalogue
Screenshot 2022-12-21 at 11 57 31 AM
#### Replenishment selected, hover over Catalogue > View Stock
Screenshot 2022-12-21 at 11 55 49 AM

### Unable to replicate 
**Site**
- The footer text is jammed together ( storename username )

### Already done
**Other components**
- Autocomplete - highlight the search terms when filtering, perhaps bold the search characters?
- Date picker - is the selected date colour the primary? it doesn't look right

### CODE REVIEW

**Comment by CarlosNZ** [comment]
>  highlighting of the parent menu when you are on a submenu page. It now has the parent in bold with the whole menu (parent+children) in a container with a box shadow. Shown as a very light grey when active, and transparent otherwise.

Screenshot 2023-01-09 at 4 00 04 PM

This one's not really working for me. I think it's the way the shadows are interacting is destroying the illusion of depth. Cos the box around the parent category makes that whole block look raised, and then you have a "raised" button on top of that too. But because the button is going right to the edge of the category box, you should really the bit of shadow that spreads onto the lower background "spread" further than it is when it's only going one layer down (which is obviously not easily done in pure CSS).

I think having the parent text in Bold is enough -- do we really need two layers of shadowed items? (there's no confusion as to which parent is selected when the button of the active item is quite clear)

(Other suggestions -- maybe the parent category can just have a light "border" around it. Also, some padding between the selected button and the surround frame might help).

**Comment by CarlosNZ** [comment]
> * when the drawer is collapsed, highlight the parent if you are on a submenu page

Good idea, but I think full white is too bright here -- it draws your eye too much to the side when it's not actually very relevant to the content you're looking at. When hovering, grey button on grey background looks fine (cos the drop shadow makes it stand out) -- could we just keep it the same grey when selected?

**Comment by CarlosNZ** [comment]
> * add an 'external link' indication to the docs nav link

Cool, but be good to have it centered vertically (or maybe the bottom could be flush with the text base):
Screenshot 2023-01-09 at 4 04 45 PM

File: client/packages/common/src/ui/components/domain/Dashboard/StatsPanel.tsx [hunk 1]
```diff
@@ -101,12 +101,6 @@ export const StatsPanel: FC<StatsPanelProps> = ({
               ))}
             </Grid>
           )}
-          <Grid item>
-            <BarChartIcon
-              sx={{ height: '50px', width: '125px' }}
-              color="secondary"
-            />
-          </Grid>
```
Comment (CarlosNZ) [inline client/packages/common/src/ui/components/domain/Dashboard/StatsPanel.tsx:109]: Yeah, good call. I actually thought this was a live chart (guess I hadn't looked very hard 😄), not just a static icon -- so definitely good to remove. :)

**Review by CarlosNZ — CHANGES_REQUESTED** [review]
Cool, code all looks fine. I just have a few comments regarding the actual look of certain elements, have posted them as comments in main chat. 

Happy to discuss in person to work out best solution, or we can wait to get Anisha's thoughts?

**Comment by mark-prins** [comment]
Have made changes, based on review feedback. @CarlosNZ thanks for the review & good advice!
@anisha-msupply are you able to have a look at the screenshots in here and see if they look ok?

The highlighted state when collapsed is more subtle now

Screenshot 2023-01-10 at 9 54 24 AM


The submenu items have chevrons to indicate the link and are indented slightly to show their child status - without the background colour on the group.

Screenshot 2023-01-10 at 9 53 57 AM

When another menu is opened, I think that it is clear enough to see which is current, and which is open:

Screenshot 2023-01-10 at 9 54 07 AM

The hover state is the same for children, though they aren't as high as the parent menu items:

Screenshot 2023-01-10 at 9 54 18 AM


**Bonus!** have moved the badge component, so that it is more closely aligned with the text:

Screenshot 2023-01-10 at 10 06 45 AM
Screenshot 2023-01-10 at 11 24 00 AM
Screenshot 2023-01-10 at 11 26 01 AM

then went a little further. I was concerned that the vertically centred, padded external link icon was inconsistent, so have turned that into a badge, using the same styles as the standard nav link badges:

Screenshot 2023-01-10 at 11 33 05 AM
Screenshot 2023-01-10 at 11 33 13 AM

**Comment by anisha-msupply** [comment]
> @anisha-msupply are you able to have a look at the screenshots in here and see if they look ok?

Looks good @mark-prins!

I agree with @CarlosNZ on the external link icon 

I do have a couple of suggestions re menu item states - the selected child item should look most prominent, followed by the parent item and then the hover state. I'm going to mock up what this could look like, or happy to discuss in person :)

**Comment by mark-prins** [comment]
Thanks for reviews 🙏 
Have reverted the external link icon changes in bc6c3276b and will create a new issue for the further menu changes

**Pull request merged**